src/lib.rs

```diff
@@ -55,7 +61,63 @@ where
             Error::Spi(spi) => write!(f, "Error::Spi({:?})", spi),
             Error::Gpio(gpio) => write!(f, "Error::Gpio({:?})", gpio),
             Error::UnexpectedStatus => f.write_str("Error::UnexpectedStatus"),
+            Error::SectorLength => f.write_str("Error::SectorLength"),
             Error::__NonExhaustive(_) => unreachable!(),
         }
     }
 }
+
+/// A trait for reading operations from a memory chip
+pub trait Read<Addr, SPI: Transfer<u8>, CS: OutputPin> {
+    /// Reads bytes from a memory chip
+    ///
+    /// # Parameters
+    /// * `addr`: The address to start reading at
+    /// * `buf`: The buffer to read buf.len() bytes into
+    fn read(&mut self, addr: Addr, buf: &mut [u8]) -> Result<(), Error<SPI, CS>>;
+}
+
+/// A trait for writing and erasing operations on a memory chip
+pub trait BlockDevice<Addr: TryInto<usize> + Copy, SPI: Transfer<u8>, CS: OutputPin>
+where
+    <Addr as core::convert::TryInto<usize>>::Error: core::fmt::Debug,
+{
+    /// The sector length in bytes, should be set to 1 for EEPROM implementations
+    const SECTOR_LENGTH: usize;
+
+    /// Erases bytes from the memory chip
+    ///
+    /// This function will return a `SectorLength` error if `amount` is not a multiple
+    /// of [SECTOR_LENGTH](BlockDevice::SECTOR_LENGTH)
+    ///
+    /// # Parameters
+    /// * `addr`: The address to start erasing at
+    /// * `amount`: The amount of bytes to erase, starting at `addr`
+    fn erase_bytes(&mut self, addr: Addr, amount: usize) -> Result<(), Error<SPI, CS>> {
+        if amount < Self::SECTOR_LENGTH
+            || amount % Self::SECTOR_LENGTH != 0
+            || addr.try_into().unwrap() % Self::SECTOR_LENGTH != 0
+        {
+            return Err(Error::SectorLength);
+        }
+        unsafe { self.erase_bytes_unchecked(addr, amount) }
+    }
+
+    /// The "internal" method called by [erase_bytes](BlockDevice::erase_bytes), this function doesn't
+    /// need to perform the checks regarding [SECTOR_LENGTH](BlockDevice::SECTOR_LENGTH) and is not supposed
+    /// to be called by the end user of this library (which is the reason it is marked unsafe)
+    unsafe fn erase_bytes_unchecked(
+        &mut self,
+        addr: Addr,
+        amount: usize,
+    ) -> Result<(), Error<SPI, CS>>;
+
+    /// Erases the memory chip fully
+    fn erase_all(&mut self) -> Result<(), Error<SPI, CS>>;
```

Comment:
It would be good to add a warning saying that this operation can take very long to the docs of this method.